File: readability.py
# bring get_string from s50
from cs50 import get_string

# get user input text
text = get_string("Text: ")

# create a length for the text
length = len(text)

# counters
letters = words = sentences = i = 0

while i < length:
    # check letters in text
    if text[i].isalpha():
        letters += 1
    
    # check words in text
    if (i == 0 and text[i] != " ") or (i != length and text[i] == " " and text[i + 1] != " "):
        words += 1
    
    # check sentences in text
    if text[i] == "." or text[i] == "?" or text[i] == "!":
        sentences += 1
    
    # increase i 
    i += 1

# use the Coleman-Liau formula
l = (letters / words) * 100
s = (sentences / words) * 100
grade = round(0.0588 * l - 0.296 * s - 15.8)

# printing out grades
if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print("Grade ", grade)

File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>

//takes user's text
//evaluate reading level
//output what grade the sentence(s) would be in

//Other function
void getgrade(float l, float w, float s);

//Main function
int main(void)
{

    float l = 0;
    float w = 1;
    float s = 0;

    //input text and get length
    string text = get_string("Text: ");
    float letters = strlen(text);

    //ASCII codes  to check for upper and lower case letters in text
    for (int i = 0; i < letters; i++)
    {
        if ((text [i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            l++;
        }

    }
    printf("%i letter(s).\n", (int) l);

    //Loop to count for words by finding the spaces
    for (int i = 0; i < letters; i++)
    {
        if (text[i] == ' ')
        {
            w++;
        }
    }
    printf("%i word(s).\n", (int) w);

    //Loop to count for sentences
    //Searches for dots, question and exclamation marks
    for (int i = 0; i < letters; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            s++;
        }

    }
    printf("%i sentence(s)\n", (int) s);

    //call function for grade
    getgrade(l, w, s);

}

void getgrade(float l, float w, float s)
{
    //calculate average for letters and sentence for 100 words
    float L = (l / w) * 100;
    float S = (s / w) * 100;

    //degin to output the US grade level for the text
    int grade = round(0.0588 * L - 0.296 * S - 15.8);

    //If grade greater than 16 or less than 1
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}
